pandas/tseries/base.py

```diff
@@ -67,7 +67,7 @@ def __contains__(self, key):
         try:
             res = self.get_loc(key)
             return np.isscalar(res) or type(res) == slice or np.any(res)
-        except (KeyError, TypeError):
+        except (KeyError, TypeError, ValueError):
```

Comment:
Yep, I first dug to this level but thought it had been a choice to treat any string as 'convertible' at first sight. I'll make another proposition soon, if you prefer.